R/generateFilterValues.R

```diff
@@ -76,32 +104,49 @@ generateFilterValuesData = function(task, method = "randomForestSRC_importance",
 
   fn = getTaskFeatureNames(task)
 
-  fval = do.call(filter[[1]]$fun, c(list(task = task, nselect = nselect), more.args[[filter[[1]]$name]]))
-
-  fval = lapply(filter, function(x) {
-    x = do.call(x$fun, c(list(task = task), nselect = nselect, more.args[[x$name]]))
-    missing.score = setdiff(fn, names(x))
-    x[missing.score] = NA_real_
-    x[match(fn, names(x))]
-  })
-
-  fval = do.call(cbind, fval)
-  colnames(fval) = method
-  types = vcapply(getTaskData(task, target.extra = TRUE)$data[fn], getClass1)
-  out = data.frame(name = row.names(fval),
-    type = types,
-    fval, row.names = NULL, stringsAsFactors = FALSE)
+  if (exists("ens.method")) {
```

Comment:
Since I do not use an explicit argument in the function for an ensemble filter method but do the checking and assignment internally, I am not aware of a more elegant way.
Probably there is and I do not know it.

Is there anything particular bad about `exists()`? Works and backed up by the `assertSubset()` check for ensemble filters in l.57

Reply:
This probably comes with a performance hit -- may not matter much as it's only in one place. You could simply set `ens.method = NA` in the main body of the function and then check whether it's `NA`.

Reply:
> You could simply set ens.method = NA in the main body of the function and then check whether it's NA

That does not sound very elegant. Do you have any reference about the performance drop of `exists`? At least style wise I would like it more than the proposed workaround.

Reply:
I don't, but in general reflection tends to be expensive.

What happens if an object called `ens.method` is defined somewhere else, outside of this method?

Reply:
I don't, but in general reflection tends to be expensive.

What happens if an object called `ens.method` is defined somewhere else, outside of this method?

Reply:
I don't, but in general reflection tends to be expensive.

What happens if an object called `ens.method` is defined somewhere else, outside of this method?

Reply:
Ok, I didn't mean to post this three times, apparently github is having issues.

Reply:
> What happens if an object called ens.method is defined somewhere else, outside of this method?

This is a valid concern, yes. In this case indeed this variable from the global env is used and we error with an unspecific message. I've added another assert check in that condition to see if the defined filter makes sense.

But that does not prevent that the variable from the global env is not chosen in the first place. Guess we really need to define it in the function body. Done in https://github.com/mlr-org/mlr/pull/2456/commits/55c67c181087dc3b9b073cbd00a0e1cf734d0594.